### ISSUE TITLE
Seek Keyframe Next/Prev Skipping Keyframes

### ISSUE THREAD

**Issue by ChicagoScanningMG** [mifi/lossless-cut#1311]
### I have a lot of issues to go through, so in order to make it easier for me to help you, I ask that you please try these things first

- [X] Try with the [newest version from GitHub](https://github.com/mifi/lossless-cut/releases/latest)
- [X] Read all the documentation at https://github.com/mifi/lossless-cut and **Known issues / Troubleshooting**
- [X] Search for your problem under [Issues](https://github.com/mifi/lossless-cut/issues) or [Discussions](https://github.com/mifi/lossless-cut/discussions)
- [X] If you have an issue only with one particular file, [use this form instead](https://github.com/mifi/lossless-cut/issues/new?assignees=&labels=&template=file-issue.yml)

### Operating System

Windows 10

### Steps to reproduce

For the record, I have experienced this issue on both Windows 10 and Mac OS 10.15.7.  If I bring in an h.264 file in either the MP4 or MKV container and I skip through the file using the Seek Next/Prev Keyframe buttons it sometime skips around 30min worth of keyframes.  If I'm seeking forward, it will skip 30min ahead, but if I then seek backward from there, it will hit all of the expected keyframes, indicating my initial seek forward did indeed skip keyframes.

I've have encountered this with files created by both Elgato Video Capture and OBS.

### Expected behavior

Seek Next/Prev Keyframe buttons should go directly to the Next or Previous keyframe respectively.  Keyframes should never be skipped.

### Actual behavior

Sometimes when Seeking through keyframes, upwards of 30min worth of keyframes are skipped.

**Comment by mifi** [comment]
possibly related #1018

**Comment by ChicagoScanningMG** [comment]
Ah, so is this likely because the keyframe buffer isn't fully loaded?

**Comment by mifi** [comment]
Yes, loading keyframes can take a long time on long videos. That's why it's only loaded near the cursor. Not sure how to best solve it, but maybe if the user is on the last keyframe of a keyframe window, then we should not allow to skip to the next keyframe...

**Comment by ChicagoScanningMG** [comment]
I like that idea.  It seems to only happen to me when I'm mashing the keyframe button.  I wouldn't mind if there was a delay before it jumped to the next key frame while it buffered more key frames.

**Comment by seekwindow** [comment]
It happens to me as well. When keep pressing Alt+Right for approaching the expected start of a video clip, it jumps to 30 minutes later suddenly (after 5 or 6 presses quickly). As same as seeking backward for the clip ending. The only mitigation is to slow down the key press. Pause 2 second after every 4 hits.

**Comment by mifi** [comment]
FYI the workaround for this is to load all the keyframes after opening a video
